Ver 96590 update developer doc (#927)
vertica/vertica-kubernetes: DEVELOPER.md

   ```shell
   docker image ls
   REPOSITORY           TAG       IMAGE ID       CREATED          SIZE
   vertica-logger       1.0.1     ec5ef9959692   5 days ago       7.45MB
   verticadb-operator   2.2.0     abb3f97a68b0   5 days ago       80.2MB
   vertica-k8s          2.2.0     46e4511cabf1   5 days ago       1.62GB
   rockylinux           9         639282825872   2 weeks ago      70.3MB
   ...
   ```

   - `vertica-k8s`: long-running container that runs the Vertica daemon. This container is designed for admintools deployments. For details about the admintools deployment image, see the [Dockerfile](./docker-vertica/Dockerfile). For details about the vcluster deployment image, see the [Dockerfile](./docker-vertica-v2/Dockerfile).
   - `verticadb-operator`: runs the VerticaDB operator and webhook. For details, see the [Dockerfile](./docker-operator/Dockerfile).
   - `vertica-logger`: runs the vlogger sidecar container that sends the contents of `vertica.log` to STDOUT. For details, see the [Dockerfile](./docker-vlogger/Dockerfile).

>    - `rockylinux: serves as the base image for the `vertica-k8s` image. The `make docker-build` command pulls the latest version each time.

Reviewer: ```suggestion
   - `rockylinux9: serves as the default base image for the `vertica-k8s` image. The `make docker-build` command pulls this image each time.
```

Author: Added the update.

## e2e Tests

> **IMPORTANT**

> > The e2e tests only run on operators that were deployed as an object.

Reviewer 2: This line should be removed. Deploy as an object is the only deployment option mentioned in this document (see this commit: https://github.com/vertica/vertica-kubernetes/commit/04390c61a1c811749a02b67b38d11d4c99bcf764).

Author: Will do
